Sucuri/sucuri-wordpress-plugin
# Improve deployment process

This PR uses [10up's WordPress Plugin Deploy Action](https://github.com/10up/action-wordpress-plugin-deploy) to simplify pushing updates to the wordpress.org repository.

This PR:
* Creates a `.gitattributes` file in order to exclude development files.
* Creates a `.wordpress-org` folder to store static assets consumed in the [plugin's page](https://wordpress.org/plugins/sucuri-scanner/).
* Creates a new workflow called **Deploy to WordPress.org**.

The **Deploy to WordPress.org** workflow will exclude the files on the `.gitattributes` file, copy everything in the `.wordpress-org` directory and put them into the SVN `assets` directory, and finally commit the changes to wordpress.org. You can follow the whole process in this [bash script](https://github.com/10up/action-wordpress-plugin-deploy/blob/develop/entrypoint.sh).

The **Deploy to WordPress.org** workflow is only going to be triggered when a new release is published.

## How to test this
1. We need to make sure we're not excluding important files in the deployment process, as this will break our users sites. You can test that we only exclude development files by using `git archive` [like this](https://github.com/10up/action-wordpress-plugin-deploy/blob/develop/entrypoint.sh#L94). You can also specify a custom path by using the `-o` flag, e.g: `git archive -o $HOME/Desktop/sucuri-scanner.zip HEAD` (please make sure you're on the right branch).

2. It's nice to see things in action, right? I created an amazing yet useless plugin and published it in the wordpress.org repository in order to go through the whole process. Here's the [GitHub repository](https://github.com/imgerson/honking-goose-2) and the [WordPress.org repository](https://wordpress.org/plugins/honking-goose/). Please see the [Actions](https://github.com/imgerson/honking-goose-2/actions) and [Releases](https://github.com/imgerson/honking-goose-2/releases) tab, and let me know if I missed anything.

Happy testing! 🎉
